### PR TITLE
Add support for RUST_LIB_BACKTRACE variable

### LINKED ISSUE #30: Use `RUST_LIB_BACKTRACE` env variable for non panic backtrace verbosity
With the upstream inclusion of Backtrace in std the ecosystem seems to be moving towards using `RUST_BACKTRACE` for panic based backtraces and `RUST_LIB_BACKTRACE` for `std::backtrace::Backtrace`.

I think this will take like 5 minutes to do so I'm gonna open a PR for this one too, in my patched version i just swapped it completely but after thinking abuot it for a sec I think its important to separate the verbosity the same way std does.

### PR DESCRIPTION
**Pull request by yaahc**
Loosely based on this part of std

https://doc.rust-lang.org/stable/src/std/backtrace.rs.html#228-247

closes #30

### CODE REVIEW

**Comment by athre0z** [comment]
If I understand the distinction  between `RUST_BACKTRACE` and `RUST_LIB_BACKTRACE` correctly, the former is used when capturing panic traces using the default panic handler and the latter is employed when a user manually captures backtraces, e.g. as context for their own errors. Is that correct? Until you created this issue I didn't even realize the latter existed, so I want to make sure I understand it correctly.

If my assumptions above are correct, shouldn't we somehow mirror this by having two distinct verbosity levels -- one for when used as a panic handler and one for printing user provided backtraces?

**Comment by yaahc** [comment]
Yes we should, I'll fix that

**Comment by yaahc** [comment]
Hmm, it seems like there are a few options here for how to move forward but I'm not really sure which one you'd prefer.

The main issue is the `BacktracePrinter` has a single verbosity level and is shared between the panic hook printing and the direct backtrace printing. Here are the solutions I can think of:

* carry around two verbosity levels, track if the printing logic is being invoked via `print_panic_info` and use the correct level
* split the printer so that we have one for panic printing and one for backtrace printing (breaking change i assume)

Additionally, I didn't notice before but you already added the ability to override the verbosity, so technically this issue isn't really required, I can get the correct verbosity level set in my crate even without merging this, so I guess the 3rd option is just to close this, though I do think it would be idea to provide some level of `RUST_LIB_BACKTRACE` support in the defaults.

**Comment by athre0z** [comment]
I'd say we go with the first option.

We'll also need a second `from_env` func -- perhaps call it `lib_from_env` or something like this. Totally non-intuitive name, but I can't come up with anything better right now. Same goes for the prefix for the second verbosity level. Input more than welcome.

**Comment by yaahc** [comment]
> We'll also need a second `from_env` func -- perhaps call it `lib_from_env` or something like this. Totally non-intuitive name, but I can't come up with anything better right now. Same goes for the prefix for the second verbosity level. Better suggestions more than welcome.

I'm gonna pass on bikeshedding internal API names, lib_from_env sounds great to me, lol

**Comment by athre0z** [comment]
Hahaha, alright, `lib_from_env` it is. :D

File: src/lib.rs [hunk 1]
```diff
@@ -73,13 +74,26 @@ pub enum Verbosity {
 impl Verbosity {
     /// Get the verbosity level from the `RUST_BACKTRACE` env variable.
     pub fn from_env() -> Self {
-        match std::env::var("RUST_BACKTRACE") {
-            Ok(ref x) if x == "full" => Verbosity::Full,
-            Ok(_) => Verbosity::Medium,
-            Err(_) => Verbosity::Minimal,
+        Self::convert_env(env::var("RUST_BACKTRACE").ok())
```
Comment (yaahc) [inline src/lib.rs:77]: Should these methods follow std's example and make some effort to cache these values?

Comment (athre0z) [reply src/lib.rs:77]: Nah, I think this is fine as-is. I'd argue that panic printers and thus verbosity instances are generally constructed only few times per app-run and it wouldn't be worth the effort.

**Pull request merged**